File: JTSArjuna/JTSArjuna-2.0_Alpha_2_1/ArjunaCommon/classes/com/arjuna/ArjunaCommon/Interface/NameService.java
/*
 * Copyright (C) 1998, 1999, 2000,
 *
 * Arjuna Solutions Limited,
 * Newcastle upon Tyne,
 * Tyne and Wear,
 * UK.  
 *
 * $Id: NameService.java,v 1.2.4.1 2000/04/25 08:35:56 nmcl Exp $
 */

package com.arjuna.ArjunaCommon.Interface;

import com.arjuna.ArjunaCommon.ArjunaCommonNames;
import com.arjuna.ArjunaCommon.Common.*;
import com.arjuna.ArjunaCommon.Implementation.NameServiceImple;

import java.io.IOException;
import java.lang.NullPointerException;

public class NameService
{

public NameService (ClassName nameServiceImpleName)
    {
	Inventory invent = Inventory.inventory();

	if (invent != null)
	{
	    Object ptr = invent.createVoid(nameServiceImpleName);
	    
	    if (ptr instanceof NameServiceImple)
		_imple = (NameServiceImple) ptr;
	    else
		_imple = null;
	}
    }

public NameService (NameService nameService)
    {
	if (nameService._imple != null)
	{
	    _imple = (NameServiceImple) nameService._imple.clone();
	}
	else
	    _imple = null;
    }

public NameService (NameServiceImple nameServiceImple)
    {
	_imple = nameServiceImple;
    }

public void finalize ()
    {
	_imple = null;
    }

public int attributeType (String objName, String attr) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.attributeType(objName, attr);
	else
	    throw new NullPointerException("Implementation not set");
    }

public String firstAttributeName (String objName) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.firstAttributeName(objName);
	else
	    throw new NullPointerException("Implementation not set");
    }

public String nextAttributeName (String objName, String attrName) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.nextAttributeName(objName, attrName);
	else
	    throw new NullPointerException("Implementation not set");
    }

public long getLongAttribute (String objName, String attrName) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.getLongAttribute(objName, attrName);
	else
	    throw new NullPointerException("Implementation not set");
    }

public String getStringAttribute (String objName, String attrName) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.getStringAttribute(objName, attrName);
	else
	    throw new NullPointerException("Implementation not set");
    }

public ObjectName getObjectNameAttribute (String objName, String attrName) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.getObjectNameAttribute(objName, attrName);
	else
	    throw new NullPointerException("Implementation not set");
    }

public ClassName getClassNameAttribute (String objName, String attrName) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.getClassNameAttribute(objName, attrName);
	else
	    throw new NullPointerException("Implementation not set");
    }

public Uid getUidAttribute (String objName, String attrName) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.getUidAttribute(objName, attrName);
	else
	    throw new NullPointerException("Implementation not set");
    }

public String setLongAttribute (String objName, String attrName, long value) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.setLongAttribute(objName, attrName, value);
	else
	    throw new NullPointerException("Implementation not set");
    }

public String setStringAttribute (String objName, String attrName, String value) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.setStringAttribute(objName, attrName, value);
	else
	    throw new NullPointerException("Implementation not set");
    }

public String setObjectNameAttribute (String objName, String attrName, ObjectName value) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.setObjectNameAttribute(objName, attrName, value);
	else
	    throw new NullPointerException("Implementation not set");
    }

public String setClassNameAttribute (String objName, String attrName, ClassName value) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.setClassNameAttribute(objName, attrName, value);
	else
	    throw new NullPointerException("Implementation not set");
    }

public String setUidAttribute (String objName, String attrName, Uid value) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.setUidAttribute(objName, attrName, value);
	else
	    throw new NullPointerException("Implementation not set");
    }

public String removeAttribute (String objName, String attrName) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.removeAttribute(objName, attrName);
	else
	    throw(new NullPointerException("Implementation not set"));
    }

public String uniqueAttributeName (String objName) throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.uniqueAttributeName(objName);
	else
	    throw(new NullPointerException("Implementation not set"));
    }

public ObjectName uniqueObjectName () throws NullPointerException, IOException
    {
	if (_imple != null)
	    return _imple.uniqueObjectName();
	else
	    throw(new NullPointerException("Implementation not set"));	    
    }

public void copy (NameService toCopy)
    {
	if (this == toCopy)
	    return;
    
	_imple = (NameServiceImple) toCopy._imple.clone();
    }
	
public static ClassName name ()
    {
	return ArjunaCommonNames.Interface_NameService();
    }

public ClassName className ()
    {
	return null;
    }

public ClassName impleClassName ()
    {
	if (_imple != null)
	    return _imple.className();
	else
	    return ClassName.invalid();
    }

public NameService castup (ClassName theType)
    {
	if (theType.equals(className()))
	    return this;
	else
	    return null;
    }
    
public static NameService create (ClassName nameServiceImpleName)
    {
	NameService res = null;
	Inventory invent = Inventory.inventory();

	if (invent != null)
	{
	    NameServiceImple nameServiceImple = null;
	    Object ptr = invent.createVoid(nameServiceImpleName);

	    if (ptr instanceof NameServiceImple)
		nameServiceImple = (NameServiceImple) ptr;
	    else
		nameServiceImple = null;
	
	    if (nameServiceImple != null)
		res = new NameService(nameServiceImple);
	}

	return res;
    }

private NameServiceImple _imple;
    
}

File: JTSArjuna/JTSArjuna-1.2.2/Gandiva/lib/Interface/NameService.cc
/*
 * Copyright (C) 1994, 1995, 1996,
 *
 * Department of Computing Science,
 * University of Newcastle upon Tyne,
 * Newcastle upon Tyne,
 * UK.
 *
 * $Id: NameService.cc,v 1.6 1996/01/02 13:34:29 nsmw Exp $
 */

#include <os/stddef.h>

#ifndef GANDIVA_H_
#   include <Gandiva.h>
#endif

#ifndef COMMON_BOOLEAN_H_
#   include <Common/Boolean.h>
#endif

#ifndef CORE_RESOURCE_H_
#   include <Core/Resource.h>
#endif

#ifndef INTERFACE_INVENTORY_H_
#   include <Interface/Inventory.h>
#endif

#ifndef IMPLEMENTATION_NAMESERVICE_H_
#   include <Implementation/NameService.h>
#endif

#ifndef INTERFACE_NAMESERVICE_H_
#   include <Interface/NameService.h>
#endif

Boolean NameService::attributeType(const char *objName, const char *attr, ObjectName::AttributeType &attrType)
{
    if (_imple != NULL)
        return _imple->attributeType(objName, attr, attrType);
    else
        return FALSE;
}

Boolean NameService::firstAttributeName(const char *objName, char *&attrName)
{
    if (_imple != NULL)
        return _imple->firstAttributeName(objName, attrName);
    else
        return FALSE;
}

Boolean NameService::nextAttributeName(const char *objName, const char *attrName, char *&nextAttrName)
{
    if (_imple != NULL)
        return _imple->nextAttributeName(objName, attrName, nextAttrName);
    else
        return FALSE;
}

Boolean NameService::getAttribute(const char *objName, const char *attrName, long &value)
{
    if (_imple != NULL)
        return _imple->getAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::getAttribute(const char *objName, const char *attrName, unsigned long &value)
{
    if (_imple != NULL)
        return _imple->getAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::getAttribute(const char *objName, const char *attrName, char *&value)
{
    if (_imple != NULL)
        return _imple->getAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::getAttribute(const char *objName, const char *attrName, ObjectName &value)
{
    if (_imple != NULL)
        return _imple->getAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::getAttribute(const char *objName, const char *attrName, ClassName &value)
{
    if (_imple != NULL)
        return _imple->getAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::getAttribute(const char *objName, const char *attrName, Uid &value)
{
    if (_imple != NULL)
        return _imple->getAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::setAttribute(char *&objName, const char *attrName, long value)
{
    if (_imple != NULL)
        return _imple->setAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::setAttribute(char *&objName, const char *attrName, unsigned long value)
{
    if (_imple != NULL)
        return _imple->setAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::setAttribute(char *&objName, const char *attrName, const char *value)
{
    if (_imple != NULL)
        return _imple->setAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::setAttribute(char *&objName, const char *attrName, const ObjectName &value)
{
    if (_imple != NULL)
        return _imple->setAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::setAttribute(char *&objName, const char *attrName, const ClassName &value)
{
    if (_imple != NULL)
        return _imple->setAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::setAttribute(char *&objName, const char *attrName, const Uid &value)
{
    if (_imple != NULL)
        return _imple->setAttribute(objName, attrName, value);
    else
        return FALSE;
}

Boolean NameService::removeAttribute(char *&objName, const char *attrName)
{
    if (_imple != NULL)
        return _imple->removeAttribute(objName, attrName);
    else
        return FALSE;
}

Boolean NameService::uniqueAttributeName(const char *objName, char *&attrName)
{
    if (_imple != NULL)
        return _imple->uniqueAttributeName(objName, attrName);
    else
        return FALSE;
}

Boolean NameService::uniqueObjectName(ObjectName &objectName)
{
    if (_imple != NULL)
        return _imple->uniqueObjectName(objectName);
    else
        return FALSE;
}

NameService& NameService::operator= (const NameService& toCopy)
{
    if (this == &toCopy)
	return *this;
    
    if (_imple)
	Resource::unref(_imple);

    if (toCopy._imple)
    {
	_imple = toCopy._imple->clone();
	Resource::ref(_imple);
    }
    else
	_imple = NULL;
    
    return *this;
}
	
const ClassName &NameService::name()
{
    return Gandiva::Interface::NameService::name();
}

const ClassName &NameService::className() const
{
    return Gandiva::Interface::NameService::name();
}

const ClassName &NameService::impleClassName() const
{
    if (_imple != NULL)
        return _imple->className();
    else
        return ClassName::invalid();
}

NameService *NameService::create(const ClassName &nameServiceImpleName)
{
    NameService *res    = NULL;
    Inventory  *invent = Inventory::inventory();

    if (invent != NULL)
    {
        Resource::ref(invent);

        NameServiceImple *nameServiceImple = NameServiceImple::castup(invent->createVoid(nameServiceImpleName));

        if (nameServiceImple != NULL)
            res = new NameService(nameServiceImple);

        Resource::unref(invent);
    }

    return res;
}

Resource *NameService::control(const ClassName &ctrlInterName)
{
    if (_imple != NULL)
        return _imple->control(ctrlInterName);
    else
        return NULL;
}

NameService *NameService::castup(Resource *resource)
{
    if (resource != NULL)
        return (NameService *) resource->castup(NameService::name());
    else
        return NULL;
}

void *NameService::castup(const ClassName &className) const
{
    if (className == NameService::name())
        return (void *) this;
    else
        return NULL;
}

NameService::NameService(const ClassName &nameServiceImpleName)
{
    Inventory *invent = Inventory::inventory();

    _imple = NULL;

    if (invent != NULL)
    {
        Resource::ref(invent);

        _imple = NameServiceImple::castup(invent->createVoid(nameServiceImpleName));
        Resource::ref(_imple);

        Resource::unref(invent);
    }
}

NameService::NameService(const NameService &nameService)
{
    if (nameService._imple)
    {
	_imple = nameService._imple->clone();
	Resource::ref(_imple);
    }
    else
	_imple = NULL;
}

NameService::NameService(NameServiceImple *nameServiceImple)
{
    _imple = nameServiceImple;
    Resource::ref(_imple);
}

NameService::~NameService()
{
    Resource::unref(_imple);
}
